src/parser/gpx.rs

```diff
@@ -5,38 +5,135 @@ use std::io::Read;
 use std::iter::Peekable;
 use xml::reader::Events;
 use xml::reader::XmlEvent;
+use geo::Bbox;
+use chrono::{ DateTime, Utc };
 
+use parser::bounds;
+use parser::time;
+use parser::string;
 use parser::track;
 use parser::metadata;
 use parser::waypoint;
 
 use Gpx;
+use GpxVersion;
+use Link;
+use Person;
+use Metadata;
 
 enum ParseEvent {
+    StartAuthor, // GPX 1.0
+    StartBounds, // GPX 1.0
+    StartDescription, // GPX 1.0
+    StartEmail, // GPX 1.0
+    StartKeywords, // GPX 1.0
     StartMetadata,
+    StartName, // GPX 1.0
+    StartTime, // GPX 1.0
     StartTrack,
+    StartUrl, // GPX 1.0
+    StartUrlname, // GPX 1.0
     StartWaypoint,
     Ignore,
     EndGpx,
 }
 
+/// Convert the version string to the version enum
+fn version_string_to_version(version_str: &str) -> GpxVersion {
+    match version_str {
+        "1.0" => GpxVersion::Gpx10,
+        "1.1" => GpxVersion::Gpx11,
+        _ => GpxVersion::Unknown
```

Comment:
Is there a requirement to have `GpxVersion::Unknown` part of the enum? Are we able to always error out if the version 1) isn't provided or 2) is malformed?

I see that the [GPX version](http://www.topografix.com/gpx/1/1/#element_gpx) is required ("You must include the version number in your GPX document."), but perhaps I'm missing something.

Reply:
Returning an error here is probably better. As for the enum, there is a purpose.